Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1290,6 +1290,10 @@ void StyleComputer::build_rule_cache()
     Vector<MatchingRule> matching_rules;
     size_t style_sheet_index = 0;
     for_each_stylesheet(CascadeOrigin::Author, [&](auto& sheet) {
+        if (!sheet.media().view().contains("screen"sv)) {
```

Comment:
This is not very reliable. "not screen" would be included, but "all" would not.

Can we use the existing media-query support instead? As far as I'm aware, it follows the same syntax and rules.

Reply:
oh, right, I checked spec and turned out `media` attribute can be equal to any valid media query. seems like I would also need to use `parse_as_media_query` from CSS parser.

Reply:
made a change to use existing media-query support and css parser but wondering if `StyleSheet` is a right class to put media query evaluator for a style because probably it should rather live in `CSSStyleSheet`.